table/table.go

```diff
@@ -173,7 +171,20 @@ func OpenTable(fd *os.File, mode options.FileLoadingMode, cksum []byte) (*Table,
 
 	switch mode {
 	case options.LoadToRAM:
-		// No need to do anything. t.mmap is already filled.
+		if _, err := t.fd.Seek(0, io.SeekStart); err != nil {
+			return nil, err
+		}
+		t.mmap = make([]byte, t.tableSize)
+		n, err := t.fd.Read(t.mmap)
+		if err != nil {
+			// It's OK to ignore fd.Close() error because we have only read from the file.
+			_ = t.fd.Close()
+			return nil, y.Wrapf(err, "Failed to load file into RAM")
+		}
+		if n != t.tableSize {
+			return nil, errors.Errorf("Failed to read all bytes from the file."+
```

Comment:
errors.Errorf instead of y.wrapf?

Reply:
No no. This is a new error. We're not wrapping an existing error.